### PR TITLE
Adds score for biggest toxins bomb

### PR DESCRIPTION
**Pull request by Ivniinvi**
## About The Pull Request
A new leaderboard for the highest toxins bomb theoretical light radius

## Why It's Good For The Game
Adds an interesting facet of competition and lasting records for the biggest bomb.

## Testing Photographs and Procedure

I tested it by blowing an orion ship and syndie bomb at the tox site, and they worked.

## Changelog
:cl:
add: There is now a leaderboard for the biggest toxins bomb (theoretical size).
/:cl:

### CODE REVIEW

File: code/datums/achievements/_achievement_data.dm [hunk 1]
```diff
@@ -71,6 +71,15 @@
 	else if(istype(A, /datum/award/score))
 		data[achievement_type] += 1
 
+/datum/achievement_data/proc/increase_score(datum/award/score/achievement_type, mob/user, value)
+	var/datum/award/score/A = SSachievements.awards[achievement_type]
+	get_data(achievement_type) //Get the current status first if necessary
+	if(A.high_scores[1] < value)
+		for(var/client/C in GLOB.clients)
+			to_chat(C, "<span class='greenannounce'><B>[user.client.key] set a new high score in [A.name]: [value]</B></span>")
```
Comment (PowerfulBacon) [inline code/datums/achievements/_achievement_data.dm]: just use to_chat(world, ...

File: code/game/machinery/doppler_array.dm [hunk 2]
```diff
@@ -183,6 +183,11 @@
 	records += R
 	//Update to viewers
 	ui_update()
+
+	for(var/mob/living/carbon/human/H in oviewers(src))
+		if(H.client)
+			INVOKE_ASYNC(H.client, /client.proc/increase_score, /datum/award/score/bomb_score, H, orig_light_range)
```
Comment (PowerfulBacon) [inline code/game/machinery/doppler_array.dm:189]: Should this be for everyone watching, or just the person doing the signaler/detonation method

Comment (Ivniinvi) [reply code/game/machinery/doppler_array.dm:189]: Bombs of sufficiently high radius are often a team effort, and I don't want to shaft anyone that helps out in favor of whoever pushes the button.

File: code/datums/achievements/_achievement_data.dm [hunk 3]
```diff
@@ -71,6 +71,15 @@
...
+	if(value > data[achievement_type])
+		data[achievement_type] = value
```
Comment (PowerfulBacon) [inline code/datums/achievements/_achievement_data.dm:80]: Is it possible that data doesn't contain achievement_type and could have an index out of bounds exception?

Comment (Ivniinvi) [reply code/datums/achievements/_achievement_data.dm:80]: No, data contains all scores, even if you don't have them.

**Comment by Ivniinvi** [comment]
forcing test rerun, Pun Pun moment

**Comment by williamcll** [comment]
Do chemical bombs work?

**Comment by Ivniinvi** [comment]
> Do chemical bombs work?

They would, because it detects any explosion, but you're not going to get into the top twenty with one of them.

**Review by PowerfulBacon — APPROVED** [review]
I have PTSD from the last time merging an achievement PR and it breaking server restarting, but the code looks fine.

**Pull request merged**